### ISSUE TITLE
Notebook: "TPUs in colab" fails

### ISSUE THREAD

**Issue by jmgc** [googlecolab/colabtools#666]
Bug report for Colab: http://colab.research.google.com/.

For questions about colab usage, please use [stackoverflow](https://stackoverflow.com/questions/tagged/google-colaboratory/).

- Describe the current behavior:

The notebook TPUs in Colab fails whe the firs cell is run. The output is: AttributeError: module 'tensorflow' has no attribute 'Session'

- Describe the expected behavior:

The expected behavior is to initialize the TPU environment.

- The web browser you are using (Chrome, Firefox, Safari, etc.):

It is independent of the browser, it is a TensorFlow code issue.

- Link to self-contained notebook that reproduces this issue
  (click the *Share* button, then *Get Shareable Link*):

https://colab.research.google.com/notebooks/tpu.ipynb

**Comment by craigcitro** [comment]
@jmgc I can't seem to repro your error -- can you share a public notebook with the full exception you're seeing?

**Comment by jmgc** [comment]
@craigcitro You are right, now it works. I do not understand a thing.

Sorry for the burden.

**Comment by craigcitro** [comment]
No worries, and thanks for the report!

If you'd switched TF versions, or installed a custom TF version, that would likely lead to this sort of bug.

**Comment by jmgc** [comment]
I was trying to use Tensor Flow 2.0 (`pip install tf-nightly-2.0-preview`) in another notebook, I do not know if they share the same "workspace".

BTW, I am trying to use the TPU in Tensor Flow 2.0. Which is the sequence that shall I follow, the ones that I have found are:

```python
TPU_WORKER = 'grpc://' + os.environ['COLAB_TPU_ADDR']
WORKER_NAME='tpu_worker'

cluster_resolver = tf.distribute.cluster_resolver.TPUClusterResolver(
    TPU_WORKER)
tf.config.experimental_connect_to_host(cluster_resolver.master())
tf.tpu.experimental.initialize_tpu_system(cluster_resolver)
tpu_strategy = tf.distribute.experimental.TPUStrategy(cluster_resolver)
tf.config.experimental_connect_to_host(TPU_WORKER, WORKER_NAME) 
```

and

```python
tpu_address = os.environ['COLAB_TPU_ADDR']
print(tpu_address)
cluster_resolver = tf.distribute.cluster_resolver.TPUClusterResolver(
        'grpc://' + tpu_address)
print(cluster_resolver)
strategy = tf.distribute.experimental.TPUStrategy(
        cluster_resolver)
```

None of them work with the current nightly preview. Both give the same error with slight changes:

```
InvalidArgumentError: Unable to find a context_id matching the specified one (2382452707061047641). Perhaps the worker was restarted, or the context was GC'd?
Additional GRPC error information:
{"created":"@1565030281.077775945","description":"Error received from peer","file":"external/grpc/src/core/lib/surface/call.cc","file_line":1039,"grpc_message":"Unable to find a context_id matching the specified one (2382452707061047641). Perhaps the worker was restarted, or the context was GC'd?","grpc_status":3}
```

and

```
RuntimeError: Unable to find a context_id matching the specified one (2382452707061047641). Perhaps the worker was restarted, or the context was GC'd?
Additional GRPC error information:
{"created":"@1565030659.900441780","description":"Error received from peer","file":"external/grpc/src/core/lib/surface/call.cc","file_line":1039,"grpc_message":"Unable to find a context_id matching the specified one (2382452707061047641). Perhaps the worker was restarted, or the context was GC'd?","grpc_status":3}
```

Can you give me a hint?

Thanks in advance

**Comment by craigcitro** [comment]
Ah -- unfortunately, TF2 + TPUs aren't yet supported. :/

**Comment by jmgc** [comment]
@craigcitro Under the risk of becoming a pain, let me add a new comment. I have made the same connection sequences that I made with tensorflow 2.0 preview, but now with the default Tensorflow (version 1.14.0), and the results have been a runtime crash, being unable to see the runtime log, and no information. In this case, the error has not been during the initialization, but while starting to use the keras model. Just add that the notebook works smoothly in a "CPU" runtime.

In this case, I understand that it can be an issue of Tensorflow, most probably something I have missed in the implementation, but as there is the "Runtime crash", just let you know.

Thanks for all your help.
